lib/takeaway.rb

```diff
@@ -0,0 +1,32 @@
+require './lib/menu'
+require './lib/message'
+class Takeaway
+  attr_reader :order, :total
+  def initialize(menu_klass=Menu.new)
+      @menu = menu_klass
+      @order = []
+      @total = 0
+  end
+
+  def list_dishes
+    @menu.dishes.map{ |dish| "#{dish[:name]}: £#{dish[:price]}"}*", "
+  end
+
+  def add(item, quantity = 1)
+    element = @menu.dishes.select{|dish| dish[:name] == item}
+    fail "That item is not available" if element[0] == nil
```

Comment:
Style/NilComparison: Prefer the use of the `nil?` predicate. (https://github.com/bbatsov/ruby-style-guide#predicate-methods)